src/context-parser.js

```diff
@@ -180,6 +180,14 @@ FastParser.prototype.walk = function(i, input, endsWithEOF) {
             if(this.tags[0].toLowerCase() === this.tags[1].toLowerCase()) {
                 reconsume = 0;  /* see 12.2.4.13 - switch state for the following case, otherwise, reconsume. */
                 this.matchEndTagWithStartTag(symbol);
+                /* 
+                  After matchEndTagWithStartTag (with start tag name == end tag name), 
+                  the state will be transition back to DATA. 
```

Comment:
whenever these 3 symbols `(StateMachine.Symbol.SPACE = 0, StateMachine.Symbol.SOLIDUS = 6, StateMachine.Symbol.GREATER = 9)` are consumed from the state 13, 16, 19, 27 (i.e. they are are RCDATA, RAWTEXT, SCRIPT end tag name state), this extra logic will be triggered and eventually, it will jump back to the DATA state, so setting `tags[0] = ''` makes sense. 

however, i am wondering whether it is the right time to reset the `tags[0] = '`. For example, if the input is `<textarea></textarea[X]> and X = `space`, then the tags[0] will be reset and jump to `before attribute name state`. should we reset when transiting to DATA state when symbol `StateMachine.Symbol.GREATER` is encountered?

and do we cover the transition from state 9 to DATA state?

thought?